### ISSUE TITLE
How can I `typeshare` a type that has a field that is not under my control?

### ISSUE THREAD

**Issue by lukaszlew** [1Password/typeshare#124]
E.g. 
```
#[typeshare]
struct Id(uuid::Uuid);
```
would not add Uuid itself to type-shared types. 

Please advise.

**Comment by czocher** [comment]
@lukaszlew

```rust
struct Id(#[typeshare(serialized_as = "string")] uuid::Uuid);
```

results in:
```typescript
export type Id = string;
```

If you generate TypeScript out of it.

**Comment by lukaszlew** [comment]
Excellent! Thank you.
Consider adding it to the documentation.

**Comment by LouisGariepy** [comment]
@lukaszlew It is documented [here](https://github.com/1Password/typeshare/blob/main/docs/src/usage/annotations.md).

But tbh it's definitely jarring that these docs are not really mentioned anywhere (at least as far as I know). Also, it seems like it uses mdbook, but the docs README doesn't really explain that fact or how to render the book locally.

I think it would be an important addition to increase the visibility of these docs. Perhaps even linking the relevant book docs in the docs.rs crate documentation (which are incredibly sparse).

**Comment by czocher** [comment]
@LouisGariepy I agree that the docs require a bit of a makeover. The TypeScript `readonly` attribute is also not mentioned at all AFAIK. There's probably a lot of other things missing too.

**Comment by lukaszlew** [comment]
So now I know how to generate annotations for types of fields that I don't control
```
#[typeshare]
struct Id(#[typeshare(serialized_as = "string")] uuid::Uuid);
```

How about if it is the wrapping struct that I don't control? E.g.:

```
#[typeshare]
type = Result<String, String>
```
won't export the `Result` itself.

**Comment by czocher** [comment]
@lukaszlew I believe you can use the newtype idiom, see an example in the tests: https://github.com/1Password/typeshare/tree/main/core/data/tests/test_optional_type_alias

and the idiom documentation: https://doc.rust-lang.org/rust-by-example/generics/new_types.html
